### PR TITLE
JSON Schema

### PR DESCRIPTION
**Pull request by geekscruff**
This PR adds a JSON Schema for the m3 specification (`m3_json_schema.json`). This can be used to validate files against the specification.

In addition:

* makes a couple of changes to the m3.yml file
* adds a rough validation script
* adds some tools info to the readme

The examples currently fail - if folks are happy with this approach I can validate and update those.

### CODE REVIEW

File: validate.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require "json_schemer"
```
Comment (dunn) [inline validate.rb]: Should we go ahead and add a Gemfile for this (and any future libraries)?

Comment (geekscruff) [reply validate.rb]: good thinking, I've submitted another commit with that in

**Comment by geekscruff** [comment]
@dunn I've changed validate.rb a little so it returns exit(true) or exit(1)

File: validate.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    puts "Schema pointer: #{error['schema_pointer']}"
+    puts "Type: #{error['type']}"
+    puts "\n"
+  end
+end
+
+if valid?
+  exit(true)
+else
+  exit(0)
```
Comment (dunn) [inline validate.rb]: This needs to be `1`, I think?

Comment (geekscruff) [reply validate.rb]: Gah, yes it does.

**Pull request merged**